include/cudaq/Optimizer/Builder/Factory.h

```diff
@@ -119,6 +119,12 @@ inline mlir::LLVM::LLVMStructType stdVectorImplType(mlir::Type eleTy) {
   return mlir::LLVM::LLVMStructType::getLiteral(ctx, eleTys);
 }
 
+inline mlir::Type stateImplType(mlir::Type eleTy) {
+  auto *ctx = eleTy.getContext();
+  auto eTy = cudaq::opt::factory::getCharType(ctx);
+  return cudaq::opt::factory::getPointerType(eTy);
```

Comment:
Can be simplified to just:
```c++
return cudaq::opt::factory::getPointerType(eleTy.getContext());
```
On the other hand, we don't use the `eleTy`, so we could erase this function and just use `getPointerType` at the call site.